lib/ui5Framework/Sapui5MavenSnapshotResolver.js

```diff
@@ -0,0 +1,235 @@
+const DIST_GROUP_ID = "com.sap.ui5.dist";
+const DIST_ARTIFACT_ID = "sapui5-sdk-dist";
+
+/**
+ * Resolver for the SAPUI5 framework
+ *
+ * This Resolver downloads and installs SNAPSHOTS of UI5 libraries from
+ * a Maven repository. It's meant for internal usage only as no use cases
+ * outside of SAP are known.
+ *
+ * @public
+ * @class
+ * @alias @ui5/project/ui5Framework/Sapui5MavenSnapshotResolver
+ * @extends @ui5/project/ui5Framework/AbstractResolver
+ */
+class Sapui5MavenSnapshotResolver extends AbstractResolver {
+	/**
+	 * @param {*} options options
+	 * @param {string} [options.snapshotEndpointUrl] Maven Repository Snapshot URL. If not provided,
+	 *	falls back to an optional <code>UI5_MAVEN_SNAPSHOT_ENDPOINT</code> environment variable,
+	 *	or the standard Maven settings.xml file (if existing).
+	 * @param {string} options.version SAPUI5 version to use
+	 * @param {boolean} [options.sources=false] Whether to install framework libraries as sources or
+	 * pre-built (with build manifest)
+	 * @param {string} [options.cwd=process.cwd()] Current working directory
+	 * @param {string} [options.ui5HomeDir="~/.ui5"] UI5 home directory location. This will be used to store packages,
+	 * metadata and configuration used by the resolvers. Relative to `process.cwd()`
+	 * @param {string} [options.cacheMode="default"] Can be "default" (cache everything, invalidate after 9 hours),
+	 * 	"off" (do not cache) and "force" (use cache only - no requests)
+	 * @param {string} [options.artifactsDir] Where to install Maven artifacts
+	 * @param {string} [options.packagesDir] Where to install packages
+	 * @param {string} [options.metadataDir] Where to store the metadata for Maven artifacts
+	 * @param {string} [options.stagingDir] The staging directory for artifacts and packages
+	 */
+	constructor(options) {
+		super(options);
+
+		const snapshotEndpointUrl =
+			options.snapshotEndpointUrl ||
+			process.env.UI5_MAVEN_SNAPSHOT_ENDPOINT;
+		const {
+			cacheMode,
+			artifactsDir,
+			packagesDir,
+			metadataDir,
+			stagingDir,
+			sources,
+		} = options;
+
+		if (!snapshotEndpointUrl) {
+			// If we resolve the settings.xml at this point, we'd need to always ask the end
+			// user for confirmation. In some cases where the resources are already cached,
+			// this is not necessary and we could skip it as a real request to the repositry won't
+			// be made.
+			this._snapshotEndpointUrlCb = Sapui5MavenSnapshotResolver._resolveSnapshotEndpointUrl;
+			// TODO: Save resolved snapshot URL when .ui5rc Configuration is implemented
+		} else {
+			this._snapshotEndpointUrlCb = () => Promise.resolve(snapshotEndpointUrl);
+		}
+
+		this._installer = new Installer({
+			ui5HomeDir: this._ui5HomeDir,
+			snapshotEndpointUrlCb: this._snapshotEndpointUrlCb,
+			cacheMode,
+			artifactsDir,
+			packagesDir,
+			metadataDir,
+			stagingDir,
+		});
+		this._loadDistMetadata = null;
+		this._sources = !!sources;
```

Comment:
`this._sources` is already set by the AbstractResolver

Reply:
Thanks